### PR TITLE
Update terms link in wallet modal + refactor unused code

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

General refactor and cleanup:
Screenshot 2024-07-02 at 18 39 43

- Refactor for unused/incorrect code
- Remove policy/legal pages and routes for the old hashbang links by redirecting to cow.fi/legal
- Updates the terms link in the wallet modal

Screenshot 2024-07-02 at 17 45 32

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/application/containers/App/RoutesApp.tsx [hunk 1]
```diff
@@ -62,9 +61,9 @@ const lazyRoutes: LazyRouteProps[] = [
   { route: RoutesEnum.FAQ_ETH_FLOW, element: <ExternalRedirect url={COWDAO_KNOWLEDGE_BASE_LINK} /> },
   { route: RoutesEnum.PLAY_COWRUNNER, element: <CowRunner /> },
   { route: RoutesEnum.PLAY_MEVSLICER, element: <MevSlicer /> },
-  { route: RoutesEnum.PRIVACY_POLICY, element: <PrivacyPolicy /> },
-  { route: RoutesEnum.COOKIE_POLICY, element: <CookiePolicy /> },
-  { route: RoutesEnum.TERMS_CONDITIONS, element: <TermsAndConditions /> },
+  { route: RoutesEnum.PRIVACY_POLICY, element: <ExternalRedirect url={COWDAO_LEGAL_LINK} /> },
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/modules/application/containers/App/RoutesApp.tsx:64]: put <ExternalRedirect to a const to not repeat

**Review by alfetopito — APPROVED** [review]

**Comment by alfetopito** [comment]
Please check the linter before merging

File: apps/cowswap-frontend/src/legacy/components/Link/index.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import React from 'react'
```
[the remark this answers is not in the document]
Comment (fairlighteth) [reply apps/cowswap-frontend/src/legacy/components/Link/index.tsx]: @alfetopito Addressed, indeed the IDE was complaining...

**Comment by fairlighteth** [comment]
> Please check the linter before merging

@alfetopito ran this, check now?

**Review by elena-zh — APPROVED** [review]
Changes LGTM!
Just to clarify: AnySwapAffectedUsers page was not removed, correct?

File: apps/cowswap-frontend/src/legacy/components/Link/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/components/Link/index.tsx]: Hehe, this is one of IDEA tricks

**Review by shoom3301 — APPROVED** [review]
Awesome!

**Pull request merged**